artiq/gateware/rtio/phy/edge_counter.py

```diff
@@ -0,0 +1,83 @@
+from migen import *
+from artiq.gateware.rtio import rtlink
+
+
+class SimpleEdgeCounter(Module):
+    """Counts rising/falling edges of an input signal.
+
+    Control (sensitivity/zeroing) is done via a single RTIO output channel,
+    which is is also used to request an input event to be emitted with the
+    current counter value.
+
+    :param input_state: The (scalar) input signal to detect edges of.
+    :param counter_width: The width of the counter register, in bits. Defaults
+        to 31 to match integers being signed in ARTIQ Python.
+    """
+
+    def __init__(self, input_state, counter_width=31):
+        assert counter_width >= 2
+
+        # RTIO interface:
+        #  - output 0: 4 bits, <count_rising><count_falling><send_event><zero_counter>
+        #  - input 0: 32 bits, accumulated edge count
+        self.rtlink = rtlink.Interface(
+            rtlink.OInterface(5, enable_replace=False),
+            rtlink.IInterface(counter_width))
+
+        #: Registered copy of the input state, in the rio clock domain.
+        self.input_state = Signal()
+
+        # # #
+
+        current_count = Signal(counter_width)
+
+        count_rising = Signal()
+        count_falling = Signal()
+        send_event_stb = Signal()
+        zero_counter_stb = Signal()
+
+        # Read configuration from RTIO output events.
+        self.sync.rio_phy += [
+            If(self.rtlink.o.stb, count_rising.eq(self.rtlink.o.data[0]),
+               count_falling.eq(self.rtlink.o.data[1]),
+               send_event_stb.eq(self.rtlink.o.data[2]),
+               zero_counter_stb.eq(self.rtlink.o.data[3])).Else(
```

Comment:
This is a case where we deviate from flake8 formatting. Align `If(` and `).Else(`.